sdk/search/perf-tests/search-documents/test/getDocumentsCount.spec.ts

```diff
@@ -0,0 +1,251 @@
+import {
+  PerfStressOptionDictionary,
+  PerfStressTest,
+  getEnvVar
+} from "@azure/test-utils-perfstress";
+import {
+  SearchClient,
+  AzureKeyCredential,
+  SearchIndexClient,
+  SearchIndex,
+  KnownAnalyzerNames
+} from "@azure/search-documents";
+import { Hotel } from "./hotel";
+
+export class GetDocumentsCountTest extends PerfStressTest {
+  public options: PerfStressOptionDictionary = {
+  };
+
+  private searchClient: SearchClient<Hotel>;
+  private searchIndexClient: SearchIndexClient;
+  private indexName: string;
+
+  constructor() {
+    super();
+    const credential = new AzureKeyCredential(getEnvVar("SEARCH_DOCUMENTS_API_KEY"));
+    const endpoint = getEnvVar("SEARCH_DOCUMENTS_ENDPOINT");
+    this.indexName = getEnvVar("SEARCH_DOCUMENTS_INDEXNAME");
+    this.searchClient = new SearchClient<Hotel>(endpoint, this.indexName, credential);
+    this.searchIndexClient = new SearchIndexClient(endpoint, credential);
```

Comment:
**Just some information..**
Perf framework provides a "parallel" option that instantiates this test class GetDocumentsCountTest "parallel" number of times.
So, with the current test, each instantiation will create a new search client and new index client per parallel instance. (If you want to test the same client across all the parallels, you can declare it outside or make it a static variable.)
Also, our core-http already does cache the HTTP agent among the clients, so it might not matter.. but just letting you know.

Coming to the globalSetup, it is run only for one of the instantiated classes.. meaning the resources created at the globalSetup are meant to be shared across the parallel instances.

Reply:
Agreed. It is intentional. I want the clients to be instantiated every time. But, globalSetup must be run only once.